### ISSUE TITLE
cron: race may cause cron jobs to never run again

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#5906]
While playing around with `flux cron` for #5905 noticed a corner case (or possibly 2).

if sync is on, and a cron job is "deferred" at the moment the user disables sync, a cron job may never run again because the defer callback will never be called again (b/c sync was disabled), thus never running that deferred cron job.  That cron job can also never be scheduled again because ...

```
int cron_entry_schedule_task (cron_entry_t *e)
{
    flux_t *h = e->ctx->h;

    /* Refuse to run more than one task at once
     */
    if (e->task) {
        flux_log (h, LOG_INFO, "cron-%ju: %s: task still running or scheduled",
                  e->id, e->name);
        return (0);
    }
    ...
}
```

`e->task` is true, so the task can never be scheduled again.  I'm not sure what the correct fix is here.  Maybe to run all the deferred jobs when sync is disabled?  i.e.

```
static int cron_entry_defer (cron_entry_t *e)                                                                                                                
{                                                                                                                                                            
    cron_ctx_t *ctx = e->ctx;                                                                                                                                
    double now = get_timestamp ();                                                                                                                           
                                                                                                                                                             
    /* If no default synchronization event or the time since the last                                                                                        
     *  sync event is very short, then run task immediately                                                                                                  
     */                                                                                                                                                      
    if (!ctx->mh || (now - ctx->last_sync) < ctx->sync_epsilon)                                                                                              
        return cron_entry_run_task (e);         
    ...
```

The `!ctx->mh` means it should have been run already, but we deferred it b/c of sync.  If we're disabling sync, we could just run all the deferred cron jobs right away?

similarly, if you re-enable sync, the event handler may never be started again b/c the event handler is only started when a task **enters** deferred state, not if a cron job is already in deferred state.  So that cron job will still never run.  This half of the fix is easy:

```
--- a/src/modules/cron/cron.c
+++ b/src/modules/cron/cron.c
@@ -516,7 +516,9 @@ static int cron_ctx_sync_event_init (cron_ctx_t *ctx, const char *topic)
         flux_log_error (ctx->h, "sync_event_init: subscribe (%s)", topic);
         return (-1);
     }
-    /* Do not start the handler until we have entries on the the list */
+    /* Do not start the handler unless we have entries on the the list */
+    if (zlist_size (ctx->deferred) > 0)
+        flux_msg_handler_start (ctx->mh);
     return (0);
 }
```

(note to self, there's a typo in the original comment "the the".)
